Handle parse error for bad head exposes list

jaredramirez opened the pull request:
This PR fixes https://github.com/roc-lang/roc/issues/7246

It improves the parse error message if you have an invalid exposes list in the module header. The new error is:
```
── WEIRD MODULE NAME in /code/proj/Main.roc ────────────────────────────────────

 I am partway through parsing a header, but I got stuck here:

    1│  module foobar []
               ^

I was expecting an `exposes` list like

    [Animal, default, tame]
```

To do this, I had to add a branch to handle `EExposes::ListStart` in the function `to_exposes_report`.

I originally wanted to add more context to this error message based on whether the header in question was an `app` or a `module`. However, that would've required adding additional metadata to `EHeader`, which seemed out of scope.

While there, I noticed the branches for `EExposes::Identified` and `EExposes::ListEnd` are the same with a TODO to split them up, so I looked into that as well. As it turns out, that would require tweaking parsing which seems out of scope, so I updated the TODO with more context and left it as is.

jaredramirez commented:
Looks like in the first CI run, a test timed out:

```
test cli_tests::test_benchmarks::base64 has been running for over 60 seconds
```

I went ahead and pulled in the latest commits from `main`, but not sure why else this test would've timed out now

smores56 commented on crates/reporting/src/error/parse.rs:
```diff
@@ -4100,6 +4100,41 @@ fn to_exposes_report<'a>(
 
         EExposes::Space(error, pos) => to_space_report(alloc, lines, filename, &error, pos),
 
+        EExposes::ListStart(pos@Position { offset: 7 }) => {
```
The `offset: 7` is a little fragile. If I write `module  xyz ...` with 2 spaces after "module", we lose this error. Can we update `EExposes` to properly track this failure?

jaredramirez replied:
Yup, good point. Fixed and also made an additional change. I updated PR description with details

jaredramirez commented:
Okay @smores56, this is now ready. I ended up undoing the change to handle `ListEnd` problems into its own branch, because it would've required tweaking some parsing code, and that felt out of scope. I added a comment in the code and updated the PR description.

I've also run all tests, formatting, etc locally and all looks good, so this _should_ pass CI now.

smores56 commented:
Thanks for the change!
